File: app/src/main/java/com/sharpdroid/registroelettronico/database/room/Converters.java
package com.sharpdroid.registroelettronico.database.room;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

public class Converters {
    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Long dateToTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }

    @TypeConverter
    public static Boolean fromInteger(Integer value) {
        return value == 1;
    }

    @TypeConverter
    public static Integer booleanToInteger(Boolean bool) {
        return bool ? 1 : 0;
    }
}

File: app/src/main/java/com/sharpdroid/registroelettronico/database/room/Converters.kt
package com.sharpdroid.registroelettronico.database.room

import android.arch.persistence.room.TypeConverter
import java.util.*

class Converters {
    @TypeConverter
    fun fromTimestamp(value: Long?): Date? = if (value == null) null else Date(value)

    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? = date?.time

    @TypeConverter
    fun fromInteger(value: Int?): Boolean = value == 1

    @TypeConverter
    fun booleanToInteger(bool: Boolean?): Int? = if (bool == true) 1 else 0
}